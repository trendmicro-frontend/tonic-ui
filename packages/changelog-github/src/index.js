/**
 * @see https://github.com/changesets/changesets/blob/main/docs/modifying-changelog-format.md
 */
import { getInfo, getInfoFromPullRequest } from '@changesets/get-github-info';

const validate = (options) => {
  if (!options?.repo) {
    throw new Error(
      'Please provide a repo to this changelog generator like this:\n"changelog": ["@tonic-ui/changelog-github", { "repo": "org/repo" }]'
    );
  }
};

const getDependencyReleaseLine = async (changesets, dependenciesUpdated, options) => {
  validate(options);

  if (dependenciesUpdated.length === 0) {
    return '';
  }

  const repo = options?.repo;
  const changesetCommits = await Promise.all(
    changesets.map(async (cs) => {
      if (cs.commit) {
        const info = await getInfo({
          repo: repo,
          commit: cs.commit,
        });
        return info?.links?.commit;
      }
      return null;
    })
  );

  const changesetLink = `- Updated dependencies [${changesetCommits.join(', ')}]`;
  const updatedDependenciesList = dependenciesUpdated.map(
    (dependency) => `  - ${dependency.name}@${dependency.newVersion}`
  );

  return [changesetLink, ...updatedDependenciesList].join('\n');
};

const getReleaseLine = async (changeset, type, options) => {
  validate(options);

  const repo = options?.repo;
  let prFromSummary = 0;
  let commitFromSummary = '';
  const usersFromSummary = [];

  const replacedChangelog = changeset.summary
    .replace(/^\s*(?:pr|pull|pull\s+request):\s*#?(\d+)/im, (_, pr) => {
      if (pr) {
        prFromSummary = Number(pr) || 0;
      }
      return '';
    })
    .replace(/^\s*commit:\s*([^\s]+)/im, (_, commit) => {
      commitFromSummary = commit;
      return '';
    })
    .replace(/^\s*(?:author|user):\s*@?([^\s]+)/gim, (_, user) => {
      usersFromSummary.push(user);
      return '';
    })
    .trim();

  const [firstLine, ...futureLines] = replacedChangelog
    .split('\n')
    .map((l) => l.trimRight());

  const links = await (async () => {
    if (prFromSummary > 0) {
      const pullRequestInfo = await getInfoFromPullRequest({
        repo: repo,
        pull: prFromSummary,
      });
      const links = pullRequestInfo.links;
      if (commitFromSummary) {
        const shortCommitId = commitFromSummary.slice(0, 7);
        return {
          ...links,
          commit: `[\`${shortCommitId}\`](https://github.com/${repo}/commit/${commitFromSummary})`,
        };
      }
      return links;
    }

    const commitToFetchFrom = commitFromSummary || changeset.commit;
    if (commitToFetchFrom) {
      const info = await getInfo({
        repo: repo,
        commit: commitToFetchFrom,
      });
      return info.links;
    }

    return {
      commit: null,
      pull: null,
      user: null,
    };
  })();

  const users = (() => {
    if (usersFromSummary.length > 0) {
      return usersFromSummary.map((userFromSummary) => {
        return `[@${userFromSummary}](https://github.com/${userFromSummary})`;
      });
    }

    if (links.user) {
      return [links.user];
    }

    return [];
  })();

  const byUsers = (users && users.length > 0) ? ` by ${users.join(', ')}` : '';

  // Only link PR or merge commit not both
  const linkPullOrCommit = (() => {
    if (links.pull) {
      return ` in ${links.pull}`;
    }
    if (links.commit) {
      return ` in ${links.commit}`;
    }
    return '';
  })();

  const prefix = '';
  const suffix = `${byUsers}${linkPullOrCommit}`;

  let returnValue = `\n\n- ${prefix}${firstLine}${suffix}`;

  if (futureLines.length > 0) {
    returnValue += `\n${futureLines.filter(Boolean).map(l => `  ${l}`).join('\n')}`;
  }

  return returnValue;
};

const changelogFunctions = {
  getDependencyReleaseLine,
  getReleaseLine,
};

export default changelogFunctions;
