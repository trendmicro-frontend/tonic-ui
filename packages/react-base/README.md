# @tonic-ui/react-base

A library of headless React UI components.

## Installation

```sh
// with npm
npm install @tonic-ui/react-base

// with yarn
yarn add @tonic-ui/react-base
```
