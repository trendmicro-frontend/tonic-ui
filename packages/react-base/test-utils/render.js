import { matchers } from '@emotion/jest';

// Add the custom matchers provided by '@emotion/jest'
expect.extend(matchers);

// re-export everything
export * from '@testing-library/react';
