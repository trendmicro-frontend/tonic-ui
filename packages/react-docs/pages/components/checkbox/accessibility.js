import { Checkbox } from '@tonic-ui/react';
import React from 'react';

const App = () => (
<Checkbox
  inputProps={{
    'aria-label': 'Label',
  }}
/>

);

export default App;