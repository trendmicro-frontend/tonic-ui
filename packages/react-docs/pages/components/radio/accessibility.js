import { Radio } from '@tonic-ui/react';
import React from 'react';

const App = () => (
<Radio
  inputProps={{
    'aria-label': 'Label',
  }}
/>
);

export default App;