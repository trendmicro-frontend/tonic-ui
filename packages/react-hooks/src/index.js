export useConst from './useConst';
export useEffectOnce from './useEffectOnce';
export useHydrated from './useHydrated';
export useIsomorphicEffect from './useIsomorphicEffect';
export useLatest from './useLatest';
export usePrevious from './usePrevious';
export useToggle from './useToggle';
