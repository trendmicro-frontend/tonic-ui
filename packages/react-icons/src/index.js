export * from './icons';
export { default as SVGIcon } from './SVGIcon';
export { default as createSVGIcon } from './utils/createSVGIcon';
