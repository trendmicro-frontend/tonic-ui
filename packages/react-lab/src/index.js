// components
export * from './date-time-pickers';
