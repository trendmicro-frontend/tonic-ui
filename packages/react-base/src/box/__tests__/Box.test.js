import { testA11y } from '@tonic-ui/react-base/test-utils/accessibility';
import { render } from '@tonic-ui/react-base/test-utils/render';
import { Box } from '@tonic-ui/react-base/src';
import React from 'react';

describe('Box', () => {
  it('should render correctly', async () => {
    const renderOptions = {
      useCSSVariables: true,
    };
    const { container } = render((
      <Box>This is a box</Box>
    ), renderOptions);

    expect(container).toMatchSnapshot();

    await testA11y(container);
  });

  it('should work correctly with the `as` prop', () => {
    const { getByText } = render(
      <Box as="a" href="www.example.com">
        Box
      </Box>
    );
    expect(getByText('Box').tagName).toBe('A');
  });
});
