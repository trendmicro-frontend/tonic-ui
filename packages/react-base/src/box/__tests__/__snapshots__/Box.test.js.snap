// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render correctly 1`] = `
<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a box
  </div>
</div>
`;
